### PR TITLE
Add loss evaluator

### PR DESCRIPTION
**Pull request by OceanPang**

### CODE REVIEW

File: mmdet/models/bbox_heads/bbox_head.py [hunk 1]
```diff
@@ -2,8 +2,8 @@
 import torch.nn as nn
 import torch.nn.functional as F
 
-from mmdet.core import (delta2bbox, multiclass_nms, bbox_target,
-                        weighted_cross_entropy, weighted_smoothl1, accuracy)
+from mmdet.core import (delta2bbox, multiclass_nms, bbox_target, accuracy)
```
Comment (hellock) [inline mmdet/models/bbox_heads/bbox_head.py]: The brackets are unnecessary in a single line.

File: mmdet/models/bbox_heads/convfc_bbox_head.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def __init__(self,
         # add reg specific branch
         self.reg_convs, self.reg_fcs, self.reg_last_dim = \
             self._add_conv_fc_branch(
-                self.num_reg_convs, self.num_reg_fcs, self.shared_out_channels)
+              self.num_reg_convs, self.num_reg_fcs, self.shared_out_channels)
```
Comment (hellock) [inline mmdet/models/bbox_heads/convfc_bbox_head.py]: This change seems weird.

File: mmdet/models/losses/focal_loss.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from mmdet.core import weighted_sigmoid_focal_loss
+
+from .cross_entropy_loss import CrossEntropyLoss
+from ..registry import LOSSES
+
+
+@LOSSES.register_module
+class FocalLoss(CrossEntropyLoss):
```
Comment (hellock) [inline mmdet/models/losses/focal_loss.py]: `FocalLoss` may directly inherit from `nn.Module`.

File: mmdet/models/losses/focal_loss.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            loss_cls = self.loss_weight * self.cls_criterion(
+                cls_score,
+                label,
+                label_weight,
+                gamma=self.gamma,
+                alpha=self.alpha,
+                *args,
+                **kwargs)
+        else:
+            raise NotImplementedError()
```
Comment (hellock) [inline mmdet/models/losses/focal_loss.py]: `NotImplementedError` does not need the parenthesis.
